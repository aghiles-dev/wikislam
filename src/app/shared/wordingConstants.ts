export interface Wording {

}

export const WordingConstants: Wording = {

}
