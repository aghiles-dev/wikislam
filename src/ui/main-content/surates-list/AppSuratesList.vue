<template>
  <div>
    <progress v-if="state.isLoading" class="progress is-small" max="100"/>

    <div v-else>
      <div class="card"
           v-for="surate in state.surates"
           :key="surate.id">
        <div class="card-content">
          <p class="is-size-4 has-text-centered has-cursor-pointer"
             @click="goToSurate(surate.id)">
            <span>{{surate.surateNumber}}. {{surate.name}} ({{surate.numberOfVerses}})({{surate.arabicName}})</span><br>
            <span class="is-size-5">{{surate.frenchName}}</span>
          </p>
        </div>
      </div>
    </div>
  </div>


</template>

<script lang="ts">
  import Vue from 'vue'
  import { Component, Prop } from 'vue-property-decorator'
  import { AppSuratesListActions, AppSuratesListProps } from './AppSuratesListConnected'
  import { AppRouteName } from '../../../config/router/AppRouteNames';

  @Component({})
  export default class AppSuratesList extends Vue {
    @Prop() state: AppSuratesListProps
    @Prop() actions: AppSuratesListActions

    mounted () {
      this.actions.fetchSuratesList()
    }

    goToSurate (surateId) {
      this.$router.push({ name: AppRouteName.SURATE_DETAILS, params: { surateId } })
    }
  }
</script>

<style lang="scss" scoped>

</style>
