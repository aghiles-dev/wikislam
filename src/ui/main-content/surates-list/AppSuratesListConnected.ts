import { bindActionCreators, Dispatch } from 'redux'
import { connect } from 'redux-vue'
import { SurateState } from '../../../app/surate/domain/SurateState'
import {
  getAllSurates,
  isSuratesListBeingFetched,
  isSuratesListFetchingInError
} from '../../../config/store/rootSelectors'
import { Wording, WordingConstants } from '../../../app/shared/wordingConstants'
import { AppState } from '../../../config/store/rootState'
import { suratesListActions } from '../../../app/surates-list/usecases/suratesList.actions'
import AppSuratesList from './AppSuratesList.vue'

export type AppSuratesListProps = {
  surates: SurateState[]
  isLoading: boolean
  isError: boolean
  wording: Wording
}

export type AppSuratesListActions = {
  fetchSuratesList: () => void
}

const mapStateToProps = (state: AppState): { state: AppSuratesListProps } => ({
  state: {
    surates: getAllSurates(state),
    isLoading: isSuratesListBeingFetched(state),
    isError: isSuratesListFetchingInError(state),
    wording: WordingConstants
  }
})

const mapDispatchToProps = (dispatch: Dispatch): { actions: AppSuratesListActions } => ({
  actions: bindActionCreators(
    {
      fetchSuratesList: suratesListActions.fetchSuratesList
    },
    dispatch
  )
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AppSuratesList)
