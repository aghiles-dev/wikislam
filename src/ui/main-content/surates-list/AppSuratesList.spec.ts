import { createLocalVue, shallowMount } from '@vue/test-utils'
import AppSuratesList from './AppSuratesList.vue'
import { AppSuratesListActions, AppSuratesListProps } from './AppSuratesListConnected'
import { WordingConstants } from '../../../app/shared/wordingConstants'

describe('Component | AppSuratesList', () => {
  let $router
  let localVue
  let state: AppSuratesListProps
  let actions: AppSuratesListActions

  beforeEach(() => {
    localVue = createLocalVue()
    $router = {
      push: jest.fn()
    }
    state = {
      surates: [],
      isLoading: false,
      isError: false,
      wording: WordingConstants
    }

    actions = {
      fetchSuratesList: jest.fn()
    }
  })

  describe('When surates are being fetched', () => {
    it('should match App snapshot', () => {
      // Given
      const wrapper = shallowMount(AppSuratesList, {
        localVue,
        mocks: { $router },
        propsData: {
          state: { ...state, isLoading: true },
          actions
        }
      })

      // Then
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('When surates have already been fetched', () => {
    it('should match App snapshot', () => {
      // Given
      const wrapper = shallowMount(AppSuratesList, {
        localVue,
        mocks: { $router },
        propsData: {
          state: {
            ...state,
            surates: [
              {
                id: 1,
                surateNumber: 1,
                beginsWithBismillah: false,
                revelationOrder: 5,
                revelationPlace: "makkah",
                name: "Al-Fatihah",
                arabicName: "الفاتحة",
                frenchName: 'L\'ouverture',
                numberOfVerses: 7
              }
            ]
          },
          actions
        }
      })

      // Then
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('on init', () => {
    it('fetches all the surates', () => {
      // When
      shallowMount(AppSuratesList, {
        localVue,
        mocks: { $router },
        propsData: { state, actions }
      })

      // Then
      expect(actions.fetchSuratesList).toHaveBeenCalledTimes(1)
    })
  })

})
