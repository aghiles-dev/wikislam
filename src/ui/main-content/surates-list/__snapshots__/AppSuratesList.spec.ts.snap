// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | AppSuratesList When surates are being fetched should match App snapshot 1`] = `
<div>
  <progress
    class="progress is-small"
    max="100"
  />
</div>
`;

exports[`Component | AppSuratesList When surates have already been fetched should match App snapshot 1`] = `
<div>
  <div>
    <div
      class="card"
    >
      <div
        class="card-content"
      >
        <p
          class="is-size-4 has-text-centered has-cursor-pointer"
        >
          <span>
            1. Al-Fatihah (7)(الفاتحة)
          </span>
          <br />
           
          <span
            class="is-size-5"
          >
            L'ouverture
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
